pkg/agent/netpol/netpol.go

```diff
@@ -126,6 +130,10 @@ func Run(ctx context.Context, nodeConfig *config.Node) error {
 	wg.Add(1)
 	go hc.RunCheck(healthCh, stopCh, &wg)
 
+	//init klog v2 which is used by kube-router
+	klog.InitFlags(nil)
+	flag.Set("v", strconv.Itoa(cmds.LogConfig.VLevel))
+
```

Comment:
The problem you're running into is that the klog flags are a process global, so all the other Kubernetes components that we run in the same process also call InitFlags and try to set the verbosity level - which results in it getting reset whenever we start another component.

This might fix your specific problem as long as the network policy controller is the last thing to be run, but it does not fix the more general issue of components fighting over the K3s process-global log level.

Reply:
I will also note that we already make the same call here:
https://github.com/k3s-io/k3s/blob/8d0255af07e95b841952563253d27b0d10bd72f0/pkg/cli/cmds/log.go#L76

Reply:
Do we need to investigate further on this or this modification is enough?
I tried to call `InitLogging()` instead of the PR command but I didn't get the logs.

Reply:
It's not so much that it's wrong as its just... there are a lot of things fighting over the klog log level. This will mostly work, but if some other component happens to be delayed a bit in startup, and changes the log level back, it will "win" and the log level won't be what you want.

A more functional fix would probably be to have the embedded executor's Bootstrap function start a goroutine that periodically sets the log level back to what we want. There will still be brief periods where other components temporarily set it to the wrong value, but at least that way it would be set to what we want **most** of the time.

https://github.com/k3s-io/k3s/blob/f2bde63eea30db7c0498aeb16b6fa09fd8cc2e0c/pkg/daemons/executor/embed.go#L48-L51